sound/soc/sof/sof-audio.h

```diff
@@ -290,6 +302,7 @@ struct snd_sof_pcm_stream {
 	 * active or not while suspending the stream
 	 */
 	bool suspend_ignored;
+	struct snd_sof_pipeline_trigger_info pipeline_info;
```

Comment:
why is this here? This is IPC4 specific-stuff that pops at this level?

Shouldn't the information on connected pipelines be stored at the pipeline level?
Also not clear what we mean by 'connected'. If there is a switch between two pipelines, does this disconnect them?

Reply:
nothing IPC4 specific here at all. Its just a list of pipelines associated with a running stream isnt it? Its used by IPC4 and ignored by IPC3. what how can I store information about connected pipelines at the pipeline level? A pipeline is simply a widget right?

Reply:
doesn't a widget have private information that can be used to store such information?

I get the argument that it doesn't hurt IPC3, but then we should also make an effort to avoid exposing new things that are IPC specific if we can.

Reply:
widget private is already used for its IPC struct payload. But even then, its not a widget properly right? i still would argue that the pipeline connections are a PCM stream property. 
And the main reason for doing this here in sof-audio (which is IPC-specific) is to avoid walking through the widgets once more to determine the trigger order when we can reuse the existing loop